services/src/main/java/org/keycloak/services/resources/admin/IdentityProviderResource.java

```diff
@@ -209,20 +209,15 @@ private void updateIdpFromRep(IdentityProviderRepresentation providerRep, RealmM
     }
 
     // return ID of IdentityProvider from realm based on internalId of this provider
-    private static String getProviderIdByInternalId(RealmModel realm, String providerInternalId) {
-        return realm.getIdentityProvidersStream().filter(p -> Objects.equals(p.getInternalId(), providerInternalId))
-                .map(IdentityProviderModel::getAlias)
-                .findFirst()
-                .orElse(null);
+    private String getProviderIdByInternalId(String providerInternalId) {
+        IdentityProviderModel identityProviderModel = session.identityProviders().getById(providerInternalId);
+        return identityProviderModel != null ? identityProviderModel.getAlias() : null;
     }
 
     // sets internalId to IdentityProvider based on alias
-    private static void lookUpProviderIdByAlias(RealmModel realm, IdentityProviderRepresentation providerRep) {
-        IdentityProviderModel identityProviderModel = realm.getIdentityProvidersStream()
-                .filter(p -> Objects.equals(p.getAlias(), providerRep.getAlias()))
-                .findFirst()
-                .orElseThrow(NotFoundException::new);
-
+    private void lookUpProviderIdByAlias(IdentityProviderRepresentation providerRep) {
```

Comment:
Wouldn't be easier to read if this method is named `getInternalIdByAlias` and returns the ID rather than setting it to the representation?

Reply:
I don't know - I haven't changed the existing logic or the names of existing methods, it's doing the same thing it was doing before. I can see if I can make this thing clearer.